lib/system/database.rb

```diff
@@ -148,22 +148,25 @@ def adapter_module
 end
 
 ActiveSupport.on_load(:active_record) do
-if System::Database.oracle? && defined?(ActiveRecord::ConnectionAdapters::OracleEnhancedAdapter::DatabaseTasks)
-  ActiveRecord::ConnectionAdapters::OracleEnhancedAdapter::DatabaseTasks.class_eval do
-    prepend(Module.new do
-
-      def create
-        super
-        connection.execute "GRANT create trigger TO #{username}"
-        connection.execute "GRANT create procedure TO #{username}"
-      end
-
-      protected
-
-      def username
-        @config['username']
-      end
-    end)
+  if System::Database.oracle? && defined?(ActiveRecord::ConnectionAdapters::OracleEnhancedAdapter::DatabaseTasks)
+    ActiveRecord::ConnectionAdapters::OracleEnhancedAdapter::DatabaseTasks.class_eval do
+      prepend(Module.new do
+        # If ORACLE_SYSTEM_PASSWORD is provided, Porta impersonates Oracle's SYSTEM user to create/update a non-SYSTEM
+        # user and grants it permissions. This is a behaviour that should preferably be used during development as
+        # the usage of Oracle's SYSTEM user grants the application more power than it should have in the first place.
+        # Alternatively, Porta should be provided with a non-SYSTEM user that has been granted the necessary
+        # permissions.
+        def create
+          if ENV['ORACLE_SYSTEM_PASSWORD'].present?
+            super
+            connection.execute "GRANT create trigger TO #{username}"
```

Comment:
As `def username` has gone, wouldn't this fail now?

Reply:
Absolutely 🤦 
Re-added in c2413431e.